tree/ntuple/v7/src/RMiniFile.cxx

```diff
@@ -1240,9 +1243,29 @@ void ROOT::Experimental::Internal::RNTupleFileWriter::WriteTFileStreamerInfo()
    // This would prepend the streamed TList with self-decription information.
    // The streamer info record is just the streamed TList.
 
+   // For classes with read rules, we additionally need to store the read rules. We only need to store them once
+   // per class.
```

Comment:
out of curiosity: how can the same class end up multiple times in the streamer info map?

Reply:
That right, the map already makes sure that this doesn't happen. Simplified accordingly.

Reply:
If we support the multiple representation per field, this also means that for a single class there will be multiple layout (and hence multiple `TStreamerInfo`) in the same `RNTuple`.